apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/json/MapJsonReader.kt

```diff
@@ -23,285 +24,330 @@ import com.apollographql.apollo3.api.json.internal.toLongExact
  *
  * @param root the root [Map] to read from
  *
- * Because it's mostly used internally the checks/verifications are very slim.
  * To read from a [okio.BufferedSource], see also [BufferedSourceJsonReader]
  */
 class MapJsonReader(val root: Map<String, Any?>) : JsonReader {
-  // Like a Map but easier to access sequentially
-  // Could be optimized
-  private class OrderedMap(val entries: List<Entry>)
-  private class Entry(val key: String, val value: Any?)
 
-  private val dataStack = arrayOfNulls<Any>(MAX_STACK_SIZE)
-  private val indexStack = IntArray(MAX_STACK_SIZE)
-  private val nameIndexStack = IntArray(MAX_STACK_SIZE)
+  private var peekedToken: JsonReader.Token = JsonReader.Token.END_OBJECT
```

Comment:
It could be set to `JsonReader.Token.BEGIN_OBJECT` here and the assignment in `init` be removed?